### ISSUE TITLE
Does downloadApolloSchema support android product flavor?

### ISSUE THREAD

**Issue by jeffxchu** [apollographql/apollo-android#2710]
**Question.**
In my project, I have set up both buildType and product flavors e.g.

```
buildTypes {
    getByName("release") {
      isMinifyEnabled = true
      proguardFiles(getDefaultProguardFile("proguard-android-optimize.txt"), "proguard-rules.pro")
      isDebuggable = false
    }
    getByName("debug") {
      applicationIdSuffix(".debug")
      versionNameSuffix("-debug")
      isDebuggable = true
    }
  }

  flavorDimensions("mode")
  productFlavors {
    create("local") {
      dimension = "mode"
      applicationIdSuffix(".local")
      versionNameSuffix("-local")
      buildConfigField("String", "GRAPHQL_API", "\"https://local.abc.io/graphql\"")
    }
    create("staging") {
      dimension = "mode"
      applicationIdSuffix(".staging")
      versionNameSuffix("-staging")
      buildConfigField("String", "GRAPHQL_API", "\"https://staging.abc.io/graphql\"")
    }
  }
```
When I ran 
```
./gradlew downloadApolloSchema --endpoint="..." --schema="..."
```

I got
```
> Task :app:downloadApolloSchema FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:downloadApolloSchema'.
> Cannot find compilation unit, check variant and service.
  Possible services:
  service
  Possible variants:
  localDebug
  localDebugAndroidTest
  localDebugUnitTest
  localRelease
  localReleaseUnitTest
```
I can't find a way to resolve this error. If I remove the product flavor {}, I am able to download the `schema.json` again. I'm wondering if there are any workarounds for this issue.

**Comment by martinbonnin** [comment]
Sorry about that, this was an oversight. `--schema` should always allow to override the compilation unit automatic detection. 

A fix is in #2712.

In the meantime, you should be able to workaround this by passing the variant explicitely:

```
./gradlew downloadApolloSchema --endpoint="..." --schema="..." --variant="localRelease"
```

**Comment by jeffxchu** [comment]
> Sorry about that, this was an oversight. `--schema` should always allow to override the compilation unit automatic detection.
> 
> A fix is in #2712.
> 
> In the meantime, you should be able to workaround this by passing the variant explicitely:
> 
> ```
> ./gradlew downloadApolloSchema --endpoint="..." --schema="..." --variant="localRelease"
> ```

great. thanks for fixing this issue @martinbonnin
